lsst/ts/cbp/statemachine.py

```diff
@@ -0,0 +1,527 @@
+        self.log.info(start_ack.ack.ack)
+
+    async def enable(self):
+        """Calls the enable command and awaits the result which logs the acknowledgement code.
+
+        Returns
+        -------
+        None
+
+        """
+        enable_topic = self.remote.cmd_enable.DataType()
+        enable_ack = await self.remote.cmd_enable.start(enable_topic,timeout=10)
+        self.log.info(enable_ack.ack.ack)
+
+    async def move_azimuth(self, azimuth):
+        """Calls the moveAzimuth command and awaits the result which logs the acknowledgement code.
+
+        Parameters
+        ----------
+        azimuth: float
+            The value of the azimuth to move to in degrees.
+
+        Returns
+        -------
+        None
+
+        """
+        move_azimuth_topic = self.remote.cmd_moveAzimuth.DataType()
+        move_azimuth_topic.azimuth = azimuth
+        move_azimuth_ack = await self.remote.cmd_moveAzimuth.start(move_azimuth_topic,timeout=10)
+        self.log.info(move_azimuth_ack.ack.ack)
+
+    async def move_altitude(self, altitude):
+        """Calls the moveAltitude command and awaits and logs the acknowledgement code.
+
+        Parameters
+        ----------
+        altitude: int
+            The altitude in degrees to move CBP to.
+
+        Returns
+        -------
+        None
+
+        """
+        move_altitude_topic = self.remote.cmd_moveAltitude.DataType()
+        move_altitude_topic.altitude = altitude
+        move_altitude_ack = await self.remote.cmd_moveAltitude.start(move_altitude_topic,timeout=10)
+        self.log.info(move_altitude_ack.ack.ack)
+
+    async def set_focus(self,focus):
+        """Sends the setFocus command and awaits and logs the acknowledgement code.
+
+        Parameters
+        ----------
+        focus: int
+            The focus in microns to set the focus encoder to.
+
+        Returns
+        -------
+        None
+
+        """
+        set_focus_topic = self.remote.cmd_setFocus.DataType()
+        set_focus_topic.focus = focus
+        set_focus_ack = await self.remote.cmd_setFocus.start(set_focus_topic,timeout=10)
+        self.log.info(set_focus_ack.ack.ack)
+
+    async def change_mask(self, mask):
+        """Calls the changeMask command and awaits and logs the acknowledgement code.
+
+        Parameters
+        ----------
+        mask: str
+            The string of the mask name to change to.
+        Returns
+        -------
+        None
+
+        """
+        change_mask_topic = self.remote.cmd_changeMask.DataType()
+        change_mask_topic.mask = mask
+        change_mask_ack = await self.remote.cmd_changeMask.start()
+        self.log.info(change_mask_ack.ack.ack)
+
+    async def park(self):
+        """Calls the park command and awaits and logs the acknowledgement code.
+
+        Returns
+        -------
+        None
+
+        """
+        park_topic = self.remote.cmd_park.DataType()
+        park_ack = await self.remote.cmd_park.start(park_topic,timeout=10)
```

Comment:
is this a reasonable timeout? I'm not sure how long it takes to park.

Reply:
The timeout are not for how long it takes, but for how long it waits until returning if no acknowledgement is returned. If it parks and acknowledges it, then it will be quicker than 10 seconds.